customer-bill/src/test/java/org/fiware/tmforum/customer_bill/CustomerBillApiIT.java

```diff
@@ -0,0 +1,106 @@
+    private final CustomerBillOnDemandApiController customerBillOnDemandApiController;
+    private final AppliedCustomerBillingRateApiController appliedCustomerBillingRateApiController;
+
+    @Test
+    void test() throws JsonProcessingException, ParseException {
+        // Test CustomerBillOnDemandApi create
+        CustomerBillOnDemandCreateVO myFancyCustomerBillOnDemandCreate =
+                getCustomerBillOnDemand("My fancy first CustomerBillOnDemand", null);
+        HttpResponse<CustomerBillOnDemandVO> myFancyCustomerBillOnDemandCreateResponse =
+                customerBillOnDemandApiController.createCustomerBillOnDemand(myFancyCustomerBillOnDemandCreate)
+                        .blockingGet();
+        assertEquals(HttpStatus.CREATED,
+                myFancyCustomerBillOnDemandCreateResponse.getStatus(),
+                "CustomerBillOnDemand should have been created.");
+        CustomerBillOnDemandVO myFancyCustomerBillOnDemand = myFancyCustomerBillOnDemandCreateResponse.body();
+
+        // Test CustomerBillOnDemand 2nd create with related party
+        RelatedPartyRefVO relatedPartyRefVO = new RelatedPartyRefVO();
+        relatedPartyRefVO.setName("My related party ref");
+        relatedPartyRefVO.setId(myFancyCustomerBillOnDemand.getId());
+        relatedPartyRefVO.setRole("My related party ref role");
+        CustomerBillOnDemandCreateVO myFancySecondCustomerBillOnDemandCreate =
+                getCustomerBillOnDemand("My fancy 2nd CustomerBillOnDemand", relatedPartyRefVO);
+        HttpResponse<CustomerBillOnDemandVO> myFancySecondCustomerBillOnDemandCreateResponse =
+                customerBillOnDemandApiController.createCustomerBillOnDemand(myFancySecondCustomerBillOnDemandCreate)
+                        .blockingGet();
+        assertEquals(HttpStatus.CREATED,
+                myFancySecondCustomerBillOnDemandCreateResponse.getStatus(),
+                "2nd CustomerBillOnDemand should have been created.");
+        CustomerBillOnDemandVO myFancySecondCustomerBillOnDemand = myFancySecondCustomerBillOnDemandCreateResponse.body();
+
+        // Test CustomerBillOnDemandApi retrieve of 2nd CustomerBillOnDemand
+        HttpResponse<CustomerBillOnDemandVO> customerBillOnDemandResponse =
+                customerBillOnDemandApiController.retrieveCustomerBillOnDemand(
+                        myFancySecondCustomerBillOnDemand.getId(), null)
+                        .blockingGet();
+        assertEquals(HttpStatus.OK,
+                customerBillOnDemandResponse.getStatus(),
+                "A CustomerBillOnDemand response is expected with status OK.");
+        assertTrue(customerBillOnDemandResponse.getBody().isPresent(),
+                "A CustomerBillOnDemand response is expected with body present.");
+        assertEquals(myFancySecondCustomerBillOnDemand,
+                customerBillOnDemandResponse.getBody().get(),
+                "The full 2nd CustomerBillOnDemand should be retrieved");
+
```

Comment:
You can create them directly to the broker api. Or (probably easier here) mock the EntitiesApi client and return such objects directly(in another test without a running broker).